### PR TITLE
[DOCS] Update local dev instructions to use `start-local`

### PR DESCRIPTION
**Pull request by leemthompo**
https://www.elastic.co/start-local is live and will be our go-to local dev setup.

This PR:

- Updates both the Elasticsearch root readme and `run-elasticsearch-locally.asciidoc`

🧹 Also try to keep as concise as possible by not mirroring _everything_ in readme

### CODE REVIEW

File: docs/reference/run-elasticsearch-locally.asciidoc [hunk 1]
```diff
@@ -9,163 +15,75 @@
...
- If you don't need access to the {kib} UI, then you can skip those instructions.
+This setup comes with a one-month trial of the Elastic *Platinum* license.
+After the trial period, the license reverts to *Free and open - Basic*.
+Refer to https://www.elastic.co/subscriptions[Elastic subscriptions] for more information.
 
 [discrete]
 [[local-dev-prerequisites]]
 === Prerequisites
 
 If you don't have Docker installed, https://www.docker.com/products/docker-desktop[download and install Docker Desktop] for your operating system.
```
Comment (shainaraskas) [inline docs/reference/run-elasticsearch-locally.asciidoc:31]:
```suggestion
- If you don't have Docker installed, https://www.docker.com/products/docker-desktop[download and install Docker Desktop] for your operating system.
```

File: docs/reference/run-elasticsearch-locally.asciidoc [hunk 2]
```diff
@@ -9,163 +15,75 @@
...
```
Comment (shainaraskas) [inline docs/reference/run-elasticsearch-locally.asciidoc:32]:
```suggestion
```

File: docs/reference/run-elasticsearch-locally.asciidoc [hunk 3]
```diff
@@ -9,163 +15,75 @@
...
-
-Configure the following environment variables.
-
-[source,sh]
-----
-export ELASTIC_PASSWORD="<ES_PASSWORD>"  # password for "elastic" username
-export KIBANA_PASSWORD="<KIB_PASSWORD>"   # Used _internally_ by Kibana, must be at least 6 characters long
-----
+- Works on Linux and macOS
+- Works on Microsoft Windows using https://learn.microsoft.com/en-us/windows/wsl/install[Windows Subsystem for Linux (WSL)]
```
Comment (shainaraskas) [inline docs/reference/run-elasticsearch-locally.asciidoc:34]:
```suggestion
- If you're using Microsoft Windows, then install https://learn.microsoft.com/en-us/windows/wsl/install[Windows Subsystem for Linux (WSL)].
```

File: docs/reference/run-elasticsearch-locally.asciidoc [hunk 4]
```diff
@@ -9,163 +15,75 @@
...
-  -e "xpack.security.http.ssl.enabled=false" \
-  -e "xpack.license.self_generated.type=trial" \
-  {docker-image}
-----
-
-[discrete]
-[[local-dev-run-kib]]
-=== Run {kib} (optional)
+* *{es}* will be running at http://localhost:9200
+* *{kib}* will be running at http://localhost:5601
```
Comment (shainaraskas) [inline docs/reference/run-elasticsearch-locally.asciidoc:53]: Suggest avoiding this construction (intro sentence should be a complete sentence).

Would prefer the endpoint info to be a little more concise. will leave this alone for now

```
* *{es}*: http://localhost:9200
* *{kib}*: http://localhost:5601
```

File: docs/reference/run-elasticsearch-locally.asciidoc [hunk 5]
```diff
@@ -9,163 +15,75 @@
...
 
-To run {kib}, you must first set the `kibana_system` password in the {es} container.
+Check the connection to Elasticsearch using `curl` in the `elastic-start-local` folder:
```
Comment (shainaraskas) [inline docs/reference/run-elasticsearch-locally.asciidoc:55]:
```suggestion
From the `elastic-start-local` folder, check the connection to Elasticsearch using `curl`:
```

File: docs/reference/run-elasticsearch-locally.asciidoc [hunk 6]
```diff
@@ -9,163 +15,75 @@
...
-  {kib-docker-image}
-----
-
-When you access {kib}, use `elastic` as the username and the password you set earlier for the `ELASTIC_PASSWORD` environment variable.
-
-[NOTE]
+[CAUTION]
 ====
-The service is started with a trial license. The trial license enables all features of Elasticsearch for a trial period of 30 days. After the trial period expires, the license is downgraded to a basic license, which is free forever.
+This setup is for local testing only. HTTPS is disabled, and Basic authentication is used for {es}. For security, {es} and {kib} are accessible only via `localhost`.
```
Comment (shainaraskas) [inline docs/reference/run-elasticsearch-locally.asciidoc:66]:
```suggestion
This setup is for local testing only. HTTPS is disabled, and Basic authentication is used for {es}. For security, {es} and {kib} are accessible only through `localhost`.
```

File: docs/reference/run-elasticsearch-locally.asciidoc [hunk 7]
```diff
@@ -9,163 +15,75 @@
...
-username = 'elastic'
-password = os.getenv('ELASTIC_PASSWORD') # Value you set in the environment variable
+The script generates a random password for the `elastic` user, which is displayed at the end of the installation and stored in the `.env` file.
 
-client = Elasticsearch(
-    "http://localhost:9200",
-    basic_auth=(username, password)
-)
+An API key for {es} is generated and stored in the `.env` file as `ES_LOCAL_API_KEY`.
+Use this key to connect to {es} with a https://www.elastic.co/guide/en/elasticsearch/client/index.html[programming language client] or the <<rest-apis,REST API>>.
```
Comment (shainaraskas) [inline docs/reference/run-elasticsearch-locally.asciidoc:76]: I prefer the original placement of this in the readme - when I originally saw the "check the connection" step I didn't understand what it was doing

File: docs/reference/run-elasticsearch-locally.asciidoc [hunk 8]
```diff
@@ -9,163 +15,75 @@
...
-  http://localhost:9200/my-new-index \
-  -H 'Content-Type: application/json'
-----
-// NOTCONSOLE
+[discrete]
+[[local-dev-additional-info]]
+=== Additional information
 
-==============
+For more detailed information about the `start-local` setup refer to the https://github.com/elastic/start-local[README on GitHub].
```
Comment (shainaraskas) [inline docs/reference/run-elasticsearch-locally.asciidoc:82]:
```suggestion
For more detailed information about the `start-local` setup, refer to the https://github.com/elastic/start-local[README on GitHub].
```

File: docs/reference/run-elasticsearch-locally.asciidoc [hunk 9]
```diff
@@ -9,163 +15,75 @@
...
-[source,python]
-----
-import os
-from elasticsearch import Elasticsearch
+[[local-dev-authentication]]
+=== Authentication
 
-username = 'elastic'
-password = os.getenv('ELASTIC_PASSWORD') # Value you set in the environment variable
+The script generates a random password for the `elastic` user, which is displayed at the end of the installation and stored in the `.env` file.
```
Comment (shainaraskas) [inline docs/reference/run-elasticsearch-locally.asciidoc:73]: this is necessary to access the services at the endpoints too ... moving up

**Review by shainaraskas — APPROVED** [review]

**Comment by shainaraskas** [comment]
@elasticmachine update branch

**Comment by leemthompo** [comment]
@elasticmachine test this

**Comment by leemthompo** [comment]
@elasticmachine update branch

**Review by ezimuel — APPROVED** [review]
LGTM!

**Comment by leemthompo** [comment]
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|8.x|[](https://github.com/elastic/elasticsearch/pull/113928)|

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sqren/backport)

**Comment by leemthompo** [comment]
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|8.15|[](https://github.com/elastic/elasticsearch/pull/113974)|

Note: Successful backport PRs will be merged automatically after passing CI.

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sqren/backport)

**Pull request merged**